operators/multiclusterobservability/controllers/multiclusterobservability/analytics/rightsizing_namespace_policy.go

```diff
@@ -0,0 +1,192 @@
+			return nil
+		}
+		log.Error(err, "Error retrieving the policy")
+		return err
+	}
+
+	// Unmarshal the inner ConfigurationPolicy
+	for _, pt := range policy.Spec.PolicyTemplates {
+		var configPolicy configpolicyv1.ConfigurationPolicy
+		err := json.Unmarshal(pt.ObjectDefinition.Raw, &configPolicy)
+		if err != nil {
+			log.Error(err, "Failed to unmarshal ConfigurationPolicy from PolicyTemplate")
+			return err
+		}
+
+		// Change ComplianceType if it's "MustOnlyHave"
+		changed := false
+		for _, objTemplate := range configPolicy.Spec.ObjectTemplates {
+			if objTemplate.ComplianceType == "MustOnlyHave" {
+				objTemplate.ComplianceType = "MustNotHave"
+				changed = true
+			}
+		}
+
+		if changed {
+			// Marshal the modified ConfigurationPolicy back into JSON
+			modifiedRaw, err := json.Marshal(configPolicy)
+			if err != nil {
+				log.Error(err, "Failed to marshal modified ConfigurationPolicy")
+				return err
+			}
+
+			pt.ObjectDefinition = runtime.RawExtension{Raw: modifiedRaw}
+		}
+	}
+
+	// Update the modified policy
+	err = c.Update(context.TODO(), policy)
+	if err != nil {
+		log.Error(err, "Failed to update the modified policy")
+		return err
+	}
+
+	log.Info("Successfully updated ComplianceType in policy")
+
+	// Wait for 1 seconds
+	time.Sleep(1 * time.Second)
+
+	return nil
+}
+
+func createOrUpdatePrometheusRulePolicy(c client.Client, prometheusRule monitoringv1.PrometheusRule) error {
+
+	policy := &policyv1.Policy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "policy.open-cluster-management.io/v1",
+			Kind:       "Policy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rsPrometheusRulePolicyName,
+			Namespace: rsNamespace,
+		},
+	}
+	log.Info("RS - Policy object created")
+
+	errPolicy := c.Get(context.TODO(), types.NamespacedName{
+		Name:      policy.Name,
+		Namespace: policy.Namespace,
+	}, policy)
+
+	if errPolicy != nil && !errors.IsNotFound(errPolicy) {
+		fmt.Println("Error retriving Policy:", errPolicy)
+		return errPolicy
+	}
+
+	// Marshal the PrometheusRule object into JSON
+	promRuleJSON, err := addAPIVersionAndKind(prometheusRule, "monitoring.coreos.com/v1", "PrometheusRule")
+	if err != nil {
+		fmt.Println("Error marshaling ConfigurationPolicy:", err)
+	}
+
+	// Define the ConfigurationPolicy object
+	configPolicy := configpolicyv1.ConfigurationPolicy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "policy.open-cluster-management.io/v1",
+			Kind:       "ConfigurationPolicy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: rsPrometheusRulePolicyConfigName,
+		},
+		Spec: &configpolicyv1.ConfigurationPolicySpec{
+			RemediationAction: "inform",
+			Severity:          "low",
+			NamespaceSelector: configpolicyv1.Target{
+				Include: []configpolicyv1.NonEmptyString{
+					configpolicyv1.NonEmptyString(rsMonitoringNamespace),
+				},
+			},
+			ObjectTemplates: []*configpolicyv1.ObjectTemplate{
+				{
+					ComplianceType: "MustOnlyHave",
```

Comment:
nit: you can use `configpolicyv1.MustOnlyHave ` as the value

Reply:
Fixed